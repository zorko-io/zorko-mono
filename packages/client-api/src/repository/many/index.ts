export * from './repository.axios.many.api.client'
export * from './repository.many.api.client'
