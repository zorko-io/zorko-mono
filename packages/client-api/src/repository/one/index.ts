export * from './repository.one.api.client'
export * from './repository.axios.one.api.client'
