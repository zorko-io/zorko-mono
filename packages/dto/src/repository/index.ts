export * from './repository'
export * from './repository.validation.schema.factory'
