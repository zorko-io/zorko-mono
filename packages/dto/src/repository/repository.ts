export interface Repository {
  id?: any;
  name: string;
  description: string;
  owner: string;
}
