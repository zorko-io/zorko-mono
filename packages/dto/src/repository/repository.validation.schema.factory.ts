import * as yup from 'yup';
import { ObjectSchema } from 'yup';

export class RepositoryValidationSchemaFactory {

  constructor() {
    this.create = this.create.bind(this);
  }

  create(): ObjectSchema {
    return  yup.object({
      id: yup.string(),
      name: yup.string().required(),
      description: yup.string().required(),
      owner: yup.string().required()
    })
  }
}

export const defaultRepositoryValidationSchemaFactory = new RepositoryValidationSchemaFactory();

export default defaultRepositoryValidationSchemaFactory;
