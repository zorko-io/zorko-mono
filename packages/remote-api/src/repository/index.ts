export * from './one'
export * from './many'
