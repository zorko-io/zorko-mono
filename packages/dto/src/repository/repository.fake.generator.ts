import * as faker from 'faker';
import { Repository } from './repository';

export class RepositoryFakeGenerator {

  getOneRandomValidRepository(): Repository {
    return {
      id: faker.random.uuid(),
      name: faker.lorem.text(5),
      description: faker.lorem.sentence(20),
      owner: faker.random.uuid()
    }
  }

  getOneWrongName(): Repository {
    return {
      id: faker.random.uuid(),
      name: '',
      description: faker.lorem.sentence(20),
      owner: faker.random.uuid()
    }
  }


  getOneWrongDescription(): Repository {
    return {
      id: faker.random.uuid(),
      name: faker.lorem.text(5),
      description: '',
      owner: faker.random.uuid()
    }
  }

  getOneMissedOwner(): Repository {
    return {
      id: faker.random.uuid(),
      name: faker.lorem.text(5),
      description: faker.lorem.sentence(20),
      owner: ''
    }
  }
}


export const repositoryFakeGenerator = new RepositoryFakeGenerator();





