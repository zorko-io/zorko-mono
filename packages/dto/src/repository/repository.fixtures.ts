import * as faker from 'faker';
import { Repository } from './repository';

export function getOneRandomValidRepository(): Repository {
  return {
    id: faker.random.uuid(),
    name: faker.lorem.text(5),
    description: faker.lorem.sentence(20),
    owner: faker.random.uuid()
  }
}

export function getOneWrongName(): Repository {
  return {
    id: faker.random.uuid(),
    name: '',
    description: faker.lorem.sentence(20),
    owner: faker.random.uuid()
  }
}

export function getOneWrongDescription(): Repository {
  return {
    id: faker.random.uuid(),
    name: faker.lorem.text(5),
    description: '',
    owner: faker.random.uuid()
  }
}

export function getOneMissedOwner(): Repository {
  return {
    id: faker.random.uuid(),
    name: faker.lorem.text(5),
    description: faker.lorem.sentence(20),
    owner: ''
  }
}



