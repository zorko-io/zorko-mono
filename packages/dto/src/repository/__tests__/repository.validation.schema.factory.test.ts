import { RepositoryValidationSchemaFactory } from '../repository.validation.schema.factory';
import { Repository } from '../repository';
import * as repositoryFixtures from '../repository.fixtures';

describe('RepositoryValidationSchemaFactory', () => {

  let schema;
  let data: Repository;
  let schemaFactory;

  beforeEach(() => {
    schemaFactory = new RepositoryValidationSchemaFactory();
    schema = schemaFactory.create();
  });

  it('passes validation for random repo', () => {
    data = repositoryFixtures.getOneRandomValidRepository();
    expect(schema.isValidSync(data)).toBeTruthy();
  });

  it('fails validation for missed name', () => {
    data = repositoryFixtures.getOneWrongName();
    expect(()=> {
      schema.validateSync(data);
    }).toThrowErrorMatchingSnapshot();
  });

  it('fails validation for missed description', () => {
    data = repositoryFixtures.getOneWrongDescription();
    expect(()=> {
      schema.validateSync(data);
    }).toThrowErrorMatchingSnapshot();
  });

  it('fails validation for missed owner', () => {
    data = repositoryFixtures.getOneMissedOwner();
    expect(() => {
      schema.validateSync(data);
    }).toThrowErrorMatchingSnapshot();
  });

});
