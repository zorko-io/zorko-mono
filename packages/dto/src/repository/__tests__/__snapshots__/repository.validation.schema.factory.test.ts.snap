// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepositoryValidationSchemaFactory fails validation for missed description 1`] = `"description is a required field"`;

exports[`RepositoryValidationSchemaFactory fails validation for missed name 1`] = `"name is a required field"`;

exports[`RepositoryValidationSchemaFactory fails validation for missed owner 1`] = `"owner is a required field"`;
